Common/src/linear_algebra/CSysSolve_b.cpp

```diff
@@ -36,27 +36,27 @@ void CSysSolve_b<ScalarType>::Solve_b(const codi::RealReverse::Real* x, codi::Re
                                       const codi::RealReverse::Real* y, const codi::RealReverse::Real* y_b, size_t n,
                                       codi::DataStore* d) {
 
-  CSysVector<su2double>* LinSysRes_b = NULL;
+  CSysVector<su2double>* LinSysRes_b = nullptr;
   d->getData(LinSysRes_b);
 
-  CSysVector<su2double>* LinSysSol_b = NULL;
+  CSysVector<su2double>* LinSysSol_b = nullptr;
   d->getData(LinSysSol_b);
 
-  CSysMatrix<ScalarType>* Jacobian = NULL;
+  CSysMatrix<ScalarType>* Jacobian = nullptr;
   d->getData(Jacobian);
 
-  CGeometry* geometry  = NULL;
+  CGeometry* geometry = nullptr;
   d->getData(geometry);
 
-  CConfig* config      = NULL;
+  const CConfig* config = nullptr;
   d->getData(config);
 
-  CSysSolve<ScalarType>* solver = NULL;
+  CSysSolve<ScalarType>* solver = nullptr;
   d->getData(solver);
 
   /*--- Initialize the right-hand side with the gradient of the solution of the primal linear system ---*/
 
-  for (unsigned long i = 0; i < n; i ++) {
+  for (unsigned long i = 0; i < n; i++) {
```

Comment:
CSysVector has overloaded =, can we just use (*LinSysSol_b) = 0.0 without the loop? Not sure if that is implemented for a scalar though

Reply:
Probably slightly better performance if the loop is fused like that.